addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js

```diff
@@ -3015,7 +3015,7 @@ export class OdooEditor extends EventTarget {
         const content = block && block.innerHTML.trim();
         if (
             block &&
-            (content === '' || content === '<br>') &&
+            content === '<br>' &&
```

Comment:
Here, it checks whether the content inside a block element is either an empty string (`''`) or just a single `<br>` element. If either of these conditions is met, it adds a placeholder. However, this logic may lead to unexpected placeholders because, in some cases, the content inside a block might not be an empty string but still not require a placeholder.

We change to `content === '<br>'` specifically checks if the content inside the block is only a single `<br>` element. This change ensures that a placeholder is added only when the block contains just a line break `<br>`, which is more restrictive and less prone to causing unwanted placeholders.